hopp/simulation/technologies/hydrogen/h2_storage/salt_cavern/salt_cavern.py

```diff
@@ -0,0 +1,158 @@
+"""
+Author: Kaitlin Brunik
+Created: 7/20/2023
+Institution: National Renewable Energy Lab
+Description: This file outputs capital and operational costs of salt cavern hydrogen storage.
+It needs to be updated to with operational dynamics.
+Costs are in 2018 USD
+
+Sources:
+    - [1] Papadias 2021: https://www.sciencedirect.com/science/article/pii/S0360319921030834?via%3Dihub
+    - [2] Papadias 2021: Bulk Hydrogen as Function of Capacity.docx documentation at hopp/hydrogen/h2_storage
+    - [3] HDSAM V4.0 Gaseous H2 Geologic Storage sheet
+"""
+
+import numpy as np
+from hopp.simulation.technologies.hydrogen.h2_transport.h2_compression import Compressor
+
+class Salt_Cavern_Storage():
+    """
+    - Costs are in 2018 USD
+    - Args:
+        - `input_dict` (dict):
+            - `"H2_storage_kg"`: "float",
+            - `"storage_duration_hrs"`: "float (optional if H2_storage_kg set)",
+            - `"flow_rate_kg_hr"`: "float (optional if H2_storage_kg set)",
+            - `"system_flow_rate"` : "float",
+            - `"model"`: "str ('papadias' or 'hdsam')",
+            - `"labor_rate"`: "float (optional, default: 37.40)",
+            - `"insurance"`: "float (optional, default: 1%)",
+            - `"property_taxes"`: "float (optional, default: 1%)",
+            - `"licensing_permits"`: "float (optional, default: 0.01%)"
+    - Returns:
+        - `salt_cavern_storage_capex_per_kg`: float - the installed capital cost per kg h2 in 2018 [USD/kg]
+        - `installed_capex`: float - the installed capital cost in 2018 [USD] (including compressor)
+        - `storage_compressor_capex`: float - the installed capital cost in 2018 for the compressor [USD]
+        - `total_opex`: float - the OPEX (annual, fixed) in 2018 excluding electricity costs [USD/kg-yr]
+        - `output_dict` (dict):
+            - `'salt_cavern_storage_capex'`: float - installed capital cost in 2018 [USD]
+            - `'salt_cavern_storage_opex'`: float - OPEX (annual, fixed) in 2018  [USD/yr]
+    """
+    expected_input_dict = {
```

Comment:
Not sure if you're using this dict elsewhere, but this is actually _really_ close to how we're doing the refactors with [attrs](https://www.attrs.org/en/stable/examples.html). We could quickly update this class to use `attrs` and simplify this code + get the typing/validation benefits